source/slang/lookup.cpp

```diff
@@ -417,7 +417,7 @@ void DoLocalLookupImpl(
             {
                 checkDecl(request.semantics, inheritanceDeclRef.decl);
 
-                auto baseType = inheritanceDeclRef.getDecl()->base.type.As<DeclRefType>();
+                auto baseType = inheritanceDeclRef.getDecl()->base.type.dynamicCast<DeclRefType>();
```

Comment:
Looks like another case where the original code should have used a canonical cast.